src/NuGet.Core/NuGet.Protocol.VisualStudio/Providers/PSSearchResourceV2Provider.cs

```diff
@@ -22,12 +22,7 @@ public PSSearchResourceV2Provider()
         {
             PSSearchResource resource = null;
             var v2repo = await GetRepository(source, token);
-
-            var searchResource = await source.GetResourceAsync<UISearchResource>(token);
-            if (searchResource != null)
-            {
-                resource = new PowerShellSearchResourceV2(v2repo);
-            }
+            resource = new PowerShellSearchResourceV2(v2repo);
```

Comment:
searchResource and resource here are not related. At least this dependency is not expressed in code.